wavelength-plugins/roomshop.js

```diff
@@ -0,0 +1,211 @@
+				display += `<td>${roomshop.items[i].desc}</td>`;
+				display += `<td>${roomshop.items[i].price} ${roomshop.items[i].price > 1 ? currencyPlural : currencyName}</td>`;
+				display += `</tr>`;
+			}
+			display += `</table></center></div>`;
+			return this.sendReplyBox(display);
+		},
+
+		purchase: "buy",
+		buy: function (target, room, user) {
+			if (!Db.roomshop.has(room.id)) return this.errorReply("Roomshop is not enabled here.");
+			let roomshop = Db.roomshop.get(room.id, {items: {}});
+			if (!roomshop.bank) return this.errorReply(`${room.title} hasn't set a bank yet.`);
+			if (!roomshop.items[toId(target)]) return this.errorReply(`The item "${target}" does not exist.`);
+			if (roomshop.bank === user.userid) return this.errorReply("Bank cannot purchase from the roomshop.");
+			let bank = roomshop.bank;
+			let cost = roomshop.items[toId(target)].price;
+			// Take payments and record payments from the user
+			Economy.readMoney(user.userid, money => {
+				if (money < cost) {
+					this.errorReply(`You do not have enough ${currencyName} to purchase ${target}.`);
+					return;
+				}
+				Economy.writeMoney(user.userid, -cost, () => {
+					Economy.logTransaction(`${user.name} bought ${target} from ${room.title}'s roomshop for ${cost} ${currencyName}${Chat.plural(cost)}.`);
+				});
+				Economy.writeMoney(bank, cost, () => {
+					Economy.logTransaction(`${user.name} bought ${target} from ${room.title}'s roomshop for ${cost} ${currencyName}${Chat.plural(cost)}.`);
+				});
+			});
+			if (FS("logs/roomshops").readdirSync()) FS("logs/roomshops").mkdirpSync();
+			FS(`logs/roomshops/roomshop_${room.id}.txt`).append(`[${new Date().toUTCString()}] ${user.name} has bought ${target} from the roomshop.\n`);
+
+			let msg = `${user.name} has purchased ${target}.`;
+
+			for (let u in room.users) {
+				if (room.auth[u] === "#") {
+					user.send(`|pm|~${room.title}'s Shop Alert|${user.getIdentity()}|${msg}`);
+				}
+			}
+
+			return this.sendReply(`You have bought "${target}" for ${cost} ${currencyName}${Chat.plural(cost)} from ${room.title}'s Room Shop.`);
+		},
+
+		logs: "transactions",
+		transactions: function (target, room, user) {
+			if (!this.can("roomshop", null, room)) return false;
+			if (!Db.roomshop.has(room.id)) return this.errorReply(`Roomshop is not enabled here.`);
+			target = toId(target);
+
+			let numLines = 15;
+			let matching = true;
+
+			if (target.match(/\d/g) && !isNaN(target)) {
+				numLines = Number(target);
+				matching = false;
+			}
+
+			let topMsg = `Displaying the last ${numLines} lines of transactions:\n`;
+			let file = `logs/roomshops/roomshop_${room.id}.txt`;
+			if (!FS(file).readIfExistsSync()) return user.popup(`No transactions.`);
+
+			FS(file).read().then(data => {
+				data = data.split('\n');
+				if (target && matching) {
+					data = data.filter(function (line) {
+						return line.toLowerCase().indexOf(target.toLowerCase()) >= 0;
+					});
```

Comment:
Pretty sure this can be simplified to `data = data.filter(line => line.toLowerCase.includes(target.toLowerCase()));`